### PR TITLE
Implement plugin status monitor

### PR DESCRIPTION
**Pull request by williamjamir**

### CODE REVIEW

File: alfasim_sdk/status.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import attr
+from attr import attrib
+from attr.validators import instance_of
+
+from alfasim_sdk._validators import check_string_is_not_empty
+
+
+@attr.s(kw_only=True)
+class ErrorMessage:
+    model_name: str = attrib(validator=[instance_of(str), check_string_is_not_empty])
```
Comment (nicoddemus) [inline alfasim_sdk/status.py]: `check_string_is_not_empty` should already use `instance_of(str)` behind the scenes, so users can simply use:

```python
model_name: str = attrib(validator=check_string_is_not_empty)
```

File: tests/test_status.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from alfasim_sdk.status import ErrorMessage
+from alfasim_sdk.status import WarningMessage
+
+
+@pytest.mark.parametrize("status", [WarningMessage, ErrorMessage])
```
Comment (nicoddemus) [inline tests/test_status.py]: I suggest using `status_class`

File: tests/test_status.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+def test_status(status):
+    out = status(model_name="Test1", message="Message From test")
+    assert out.model_name == "Test1"
+    assert out.message == "Message From test"
+
+    with pytest.raises(ValueError, match='The field "model_name" cannot be empty'):
```
Comment (nicoddemus) [inline tests/test_status.py]: TBH I don't see much point in this test, you are basically testing `attr.s` functionality...

File: alfasim_sdk/hook_specs_gui.py [hunk 4]
```diff
@@ -19,6 +19,15 @@ def alfasim_get_additional_variables():
     """
 
 
+@hookspec
+def alfasim_get_status(ctx):
+    """
+    Allows plugins to execute custom checks on ALFAsim.
```
Comment (nicoddemus) [inline alfasim_sdk/hook_specs_gui.py]: One of the most important things we should document is inputs and expected outputs... feel free to use type annotations here.

File: tests/test_status.py (hunk 3 above)
Comment (williamjamir) [reply tests/test_status.py]: Yep, basically, I'm just testing that the declaration has a validator and no one has removed/overwritten by mistake or something similar.

**Comment by williamjamir** [comment]
> Just realized that a better name is `non_empty_str`, as it better matches the other validators (a noun rather than a verb)

Yep, `non_empty_str` sounds way better =)

What do you think about to change `check_unit_is_valid` to `valid_unit`?

**Comment by nicoddemus** [comment]
> What do you think about to change check_unit_is_valid to valid_unit?

Sounds good!

File: tests/test_types.py [hunk 5]
```diff
@@ -195,7 +195,7 @@ def test_tooltips():
     field = Boolean(value=True, caption="caption", tooltip="Test123")
     assert field.tooltip is "Test123"
 
-    expected_msg = "'tooltip' must be <class 'str'> \(got 2 that is a <class 'int'>\)."
+    expected_msg = r"'tooltip' must be <class 'str'> \(got 2 that is a <class 'int'>\)."
```
Comment (nicoddemus) [inline tests/test_types.py]:
```suggestion
    expected_msg = re.escape(r"'tooltip' must be <class 'str'> (got 2 that is a <class 'int'>).")
```

Comment (williamjamir) [reply tests/test_types.py]: Commit 00b3a1b applies your suggestion ;)

**Comment by nicoddemus** [comment]
Good work!

**Pull request merged**